modules/EvseV2G/v2g_ctx.cpp

```diff
@@ -462,27 +457,22 @@ bool add_service_to_service_list(struct v2g_context* v2g_ctx, const struct iso2_
     uint8_t write_idx = 0;
     bool service_found = false;
 
-    /* Try to find service in service list */
-    for (uint8_t idx = 0; idx < v2g_ctx->evse_v2g_data.evse_service_list_len; idx++) {
-        if (v2g_ctx->evse_v2g_data.evse_service_list[idx].ServiceID == evse_service.ServiceID) {
-            write_idx = idx;
+    for (const auto& service : v2g_ctx->evse_v2g_data.evse_service_list) {
+        if (service.ServiceID == evse_service.ServiceID) {
             service_found = true;
             break;
         }
+        write_idx++;
     }
 
-    if (service_found == false && (v2g_ctx->evse_v2g_data.evse_service_list_len < iso2_ServiceType_8_ARRAY_SIZE)) {
-        write_idx = v2g_ctx->evse_v2g_data.evse_service_list_len;
-        v2g_ctx->evse_v2g_data.evse_service_list_len++;
-    } else if (v2g_ctx->evse_v2g_data.evse_service_list_len == iso2_ServiceType_8_ARRAY_SIZE) {
+    if (service_found == false and (v2g_ctx->evse_v2g_data.evse_service_list.size() < iso2_ServiceType_8_ARRAY_SIZE)) {
+        v2g_ctx->evse_v2g_data.evse_service_list.push_back(evse_service);
+    } else if (v2g_ctx->evse_v2g_data.evse_service_list.size() == iso2_ServiceType_8_ARRAY_SIZE) {
         dlog(DLOG_LEVEL_ERROR, "Maximum service list size reached. Unable to add service ID %u",
              evse_service.ServiceID);
         return false;
     }
 
-    // Write service to the service list
-    v2g_ctx->evse_v2g_data.evse_service_list[write_idx] = evse_service;
```

Comment:
Now, if the service already exists, nothing is updated and this changed the existing behavior. Is this intended?